plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -165,18 +165,17 @@ func (cache *schedulerCache) ForgetPod(pod *v1.Pod) error {
 		return fmt.Errorf("pod %v state was assumed on a different node", key)
 	}
 
-	switch {
 	// Only assumed pod can be forgotten.
-	case ok && cache.assumedPods[key]:
-		err := cache.removePod(pod)
-		if err != nil {
-			return err
-		}
-		delete(cache.assumedPods, key)
-		delete(cache.podStates, key)
-	default:
+	if !ok || !cache.assumedPods[key] {
```

Comment:
Sorry - I don't understand. There isn't anything wrong with switch with single case and default - if you go over our codebase you will find a number of places with that.
I also don't understand your concern about "when it would return an error, when it would go on.".
"return" is returning from function. Your code is doing exactly the same, but for consistency with the rest of your file, I ask you to revert this part of the PR.